docs/reference/docs/reindex.asciidoc

```diff
@@ -504,6 +504,10 @@ Valid values: `index`, `create`. Defaults to `index`.
 (Optional, enum) The script language: `painless`, `expression`, `mustache`, `java`. 
 For more information, see <<modules-scripting>>.
 
+NOTE: Types in source indices are always ignored, also when not specifying a
+destination `type`. If explicitly specifying destination `type`, the specified
+type must match the type in the destination index or be either unspecified or
+the special value `_doc`. See <<removal-of-types>> for further details.
```

Comment:
I think there are two problems here:

1. The `type` parameters aren't documented.
2. It's not clear that this is chiefly for users trying to preserve their existing document types.

I pushed f6690094b1ff3d4aa0f64d71b9800e0fffbbf071 and a8087be9f4a237e30fa42545142b2225b129fa0b to this PR to address both of those.

Feel free to change or revert this commits if wanted. Thanks!

Reply:
I think I agree to 1, moving it to a definition of the `type` parameter makes sense.

About 2, while they might be trying to preserve their types, they were in reality just expecting an error in this case and I wanted to clarify that the source type is disregarded completely now. In older versions, ES would copy over the source type to the destination too, but now that dest have only one type (which in 7.x can still be freely specified) copying over the type makes little to no sense. The source could in principle still have multiple types if specifying multiple indices with different types.

I commented on this inline too.